### PR TITLE
prefer useQuery to fetch facilities for menu

### PR DESCRIPTION
**Pull request by cainlevy**
This cleanup will eventually unblock a fix for the DOC logout behavior that @jackofsometrade discovered in https://app.shortcut.com/ameelio/story/4401/providers-sees-video-even-when-needs-monitor-true#activity-4490.

It's also really satisfying to remove a Redux module and the transitionary `urql` client.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -63,32 +60,30 @@ function App({ history }: { history: History }) {
...
-  useEffect(() => {
-    if (selected) {
-      setIsInitingData(true);
-      // TODO: this initialization strategy doesn't actually work - dispatch doesnt await for the API call results
-      // kinda works but for the wrong reasons (wont work for slower API resposes)
-      // https://github.com/Ameelio/connect-doc-client/issues/74
-      (async () => {
-        await Promise.allSettled([
-          dispatch(fetchContacts()),
-          dispatch(fetchStaff()),
```
Comment (cainlevy) [inline src/App.tsx:86]: `fetchStaff` was a wasted network request. the current staff directory page loads data from graphql on mount.

File: src/App.tsx [hunk 2]
```diff
@@ -63,32 +60,30 @@ function App({ history }: { history: History }) {
...
 
-  useEffect(() => {
-    if (selected) {
-      setIsInitingData(true);
-      // TODO: this initialization strategy doesn't actually work - dispatch doesnt await for the API call results
-      // kinda works but for the wrong reasons (wont work for slower API resposes)
-      // https://github.com/Ameelio/connect-doc-client/issues/74
-      (async () => {
-        await Promise.allSettled([
-          dispatch(fetchContacts()),
```
Comment (cainlevy) [inline src/App.tsx:85]: `fetchContacts` was an over-eager network request. the visitor page dispatches `fetchContacts` on mount.

File: src/App.tsx [hunk 3]
```diff
@@ -63,32 +60,30 @@ function App({ history }: { history: History }) {
...
+    (sp) => sp.facility
+  );
 
+  // select the first facility
   useEffect(() => {
-    if (isAuthenticated) dispatch(fetchFacilities());
-  }, [isAuthenticated, dispatch]);
+    if (!facilities) return;
+    if (facilities.length === 0)
+      throw new Error("Must have access to at least one facility");
```
Comment (wittejm) [inline src/App.tsx:76]: ooh looks like this resolves this ticket? https://app.shortcut.com/ameelio/story/4446/prevent-staff-accounts-with-no-active-positions-from-logging-in

**Review by wittejm — APPROVED** [review]

File: src/App.tsx (hunk 3 above)
Comment (cainlevy) [reply src/App.tsx:76]: Yeah it might. I copied this from some redux logic but haven't verified the behavior locally yet.

**Pull request merged**